ReactiveXComponent/RabbitMq/RabbitMqSubscriber.cs

```diff
@@ -0,0 +1,237 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.IO;
+using RabbitMQ.Client;
+using RabbitMQ.Client.Events;
+using System.Reactive.Linq;
+using RabbitMQ.Client.Exceptions;
+using ReactiveXComponent.Common;
+using ReactiveXComponent.Connection;
+using ReactiveXComponent.Configuration;
+using ReactiveXComponent.Serializer;
+
+namespace ReactiveXComponent.RabbitMq
+{
+    public class RabbitMqSubscriber : IXCSubscriber
+    {
+        private bool _disposed;
+        private readonly IConnection _connection;
+        private readonly IXCConfiguration _xcConfiguration;
+        private readonly string _privateCommunicationIdentifier;
+
+        private readonly ConcurrentDictionary<SubscriberKey, RabbitMqSubscriberInfos> _subscribers;
+        private readonly ConcurrentDictionary<SubscriberKey, ThreadSafeList<Action<MessageEventArgs>>> _listenerBySubscriberKeyRepo;
+
+        public event EventHandler<MessageEventArgs> MessageReceived;
```

Comment:
If this event's use is only internal to the class or library you should change its access to private or internal.